typescriptSrc/interpreter.ts

```diff
@@ -385,6 +369,66 @@ module interpreter {
             vms.getEval().popFromVarStack() ; }
     }
 
+    function objectStepper(vms : VMS) : void {
+      // We must step the node twice. Once on a previsit and once on a postvisit.
+      if( ! vms.hasExtraInformation() ) {          
+          previsitNode(vms);
+      }
+      else {
+          // Postvisit.
+          const value = (vms.getStack() as NonEmptyVarStack).getTop();
+          vms.finishStep( value );
+          vms.getEval().popFromVarStack() ; }
+    }
+
+    function previsitNode(vms: VMS) {
+      // Previsit. Build and push stack frame
+      const manager = vms.getTransactionManager() ;
+      const stackFrame = new ObjectV( manager ) ;
+      const node = vms.getPendingNode() ;
+      const sz = node.count() ;
+      const names = new Array<string>() ;
+      for( let i=0; i < sz ; ++i ) {
+          const childNode = node.child(i) ;
+          if( childNode.label() instanceof labels.VarDeclLabel ) {
+              const name : string = childNode.child(0).label().getVal() ;
+              const initialValue = null ;
+              const type : Type = Type.NOTYPE ;
+              const field = new Field( name, NullV.theNullValue, type, false, false, manager ) ;
+              if( names.some( (v : string) => v===name ) ) {
+                  vms.reportError( "Variable " +name+ " is declared twice." ) ;
+                  return ;
+              } else {
+                  stackFrame.addField( field ) ;
+                  names.push( name ) ; }
+          }
+      } // end for
+      vms.getEval().pushOntoVarStack( stackFrame ) ;
+      // Now map this node to say it's been previsited.
+      vms.putExtraInformation( stackFrame ) ;
+      vms.setReady( false ) ;
+    }
+
+    function accessorStepper(vms: VMS) {
+      const node = vms.getPendingNode();
+      const object = vms.getChildVal(0);
+      const field = vms.getChildVal(1);
+      if (object instanceof ObjectV) {
+        if (field instanceof StringV) {
+          const val = object.getField(field.getVal()).getValue();
```

Comment:
Missing a check object.hasField( field.getVal()).  If this is true, go ahead.  If false, report an error "No field named "+field.getVal()